internal/extensions/registry.go

```diff
@@ -0,0 +1,30 @@
+package extensions
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/lunarway/shuttle/internal/global"
+)
+
+// Registry represents some kind of upstream registry where extension metadata lives, such as which ones should be downloaded, which versions they're on and how to download them
+type Registry interface {
```

Comment:
I am not sure we want to export and return the interface from here. Normally we accept interfaces and return structs?